test/common/http/http2/http2_connection_fuzz_test.cc

```diff
@@ -0,0 +1,314 @@
+      auto f = h2frame.settings();
+      for (auto& setting : f.settings()) {
+        uint16_t id = static_cast<uint16_t>(setting.identifier() & 0xffff);
+        uint32_t value = setting.value();
+        payload.append(reinterpret_cast<char*>(&id), sizeof(id));
+        payload.append(reinterpret_cast<char*>(&value), sizeof(value));
+      }
+    } else if (h2frame.has_push_promise()) {
+      type = 5;
+      auto f = h2frame.push_promise();
+      uint8_t padding_len = static_cast<uint8_t>(f.padding().size() & 0xff);
+      payload.push_back(static_cast<char>(padding_len));
+      uint32_t stream_id = f.stream_id();
+      payload.append(reinterpret_cast<char*>(&stream_id), sizeof(stream_id));
+      deflate_headers(deflater, f.headers(), payload);
+      payload.append(f.padding().c_str(), padding_len);
+    } else if (h2frame.has_ping()) {
+      type = 6;
+      auto f = h2frame.ping();
+      uint64_t pdata = f.data();
+      payload.append(reinterpret_cast<char*>(&pdata), sizeof(pdata));
+    } else if (h2frame.has_go_away()) {
+      type = 7;
+      auto f = h2frame.go_away();
+      uint32_t last_stream_id = f.last_stream_id();
+      payload.append(reinterpret_cast<char*>(&last_stream_id), sizeof(last_stream_id));
+      uint32_t error_code = f.error_code();
+      payload.append(reinterpret_cast<char*>(&error_code), sizeof(error_code));
+      payload.append(f.debug_data());
+    } else if (h2frame.has_window_update()) {
+      type = 8;
+      auto f = h2frame.window_update();
+      uint32_t wsi = f.window_size_increment();
+      payload.append(reinterpret_cast<char*>(&wsi), sizeof(wsi));
+    } else if (h2frame.has_continuation()) {
+      type = 9;
+      auto f = h2frame.continuation();
+      deflate_headers(deflater, f.headers(), payload);
+    } else {
+      // Empty frame
+      return Http2Frame("");
+    }
+
+    Http2Frame result(streamid, type, flags);
+    result.appendData(payload);
+    result.adjustPayloadSize();
+    return result;
+  } else if (pb_frame.has_junk()) {
+    // Junk frame
+    auto junk = pb_frame.junk();
+    absl::string_view contents = junk.data();
+    return Http2Frame(contents);
+  } else {
+    // Empty frame
+    return Http2Frame("");
+  }
+}
+
+envoy::config::core::v3::Http2ProtocolOptions http2Settings() {
+  envoy::config::core::v3::Http2ProtocolOptions options(
+      ::Envoy::Http2::Utility::initializeAndValidateOptions(
+          envoy::config::core::v3::Http2ProtocolOptions()));
+
+  options.mutable_hpack_table_size()->set_value(MAX_HD_TABLE_BUF_SIZE);
+  options.set_allow_metadata(true);
+  return options;
+}
+
+class Http2Harness {
+public:
+  Http2Harness() : server_settings(http2Settings()), client_settings(http2Settings()) {
+    ON_CALL(mock_server_callbacks, newStream(_, _))
+        .WillByDefault(Invoke(
+            [&](ResponseEncoder&, bool) -> RequestDecoder& { return orphan_request_decoder; }));
+  }
+
+  void fuzz_response(std::vector<Http2Frame>& frames) {
```

Comment:
Will this be used by (a) future test case(s)?

Reply:
I removed the dead code for now.